beacon-chain/rpc/eth/beacon/state.go

```diff
@@ -148,6 +149,55 @@ func (bs *Server) GetFinalityCheckpoints(ctx context.Context, req *ethpb.StateRe
 	}, nil
 }
 
+// GetRandao fetches the RANDAO mix for the requested epoch from the state identified by state_id.
+// If an epoch is not specified then the RANDAO mix for the state's current epoch will be returned.
+// By adjusting the state_id parameter you can query for any historic value of the RANDAO mix.
+// Ordinarily states from the same epoch will mutate the RANDAO mix for that epoch as blocks are applied.
+func (bs *Server) GetRandao(ctx context.Context, req *eth2.RandaoRequest) (*eth2.RandaoResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "beacon.GetRandao")
+	defer span.End()
+
+	var (
+		st  state.BeaconState
+		err error
+	)
+
+	st, err = bs.StateFetcher.State(ctx, req.StateId)
+	if err != nil {
+		return nil, helpers.PrepareStateFetchGRPCError(err)
+	}
+
+	var randao []byte
+	if req.Epoch == nil {
```

Comment:
The spec says `Epoch to fetch the RANDAO mix for. Default: state's current epoch.`. Can you add a comment explaining that in this conditional?

I think `randao_mixes` is a circular buffer, not an appended list, so the correct way to get the current randao mix is to compute the offset using the epoch of the state mod buffer size, not grabbing the end of the slice, see: 
https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#get_randao_mix